chromium_src/chrome/browser/password_manager/android/login_db_deprecation_runner_factory.cc

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/password_manager/android/login_db_deprecation_runner_factory.h"
+
+#define BuildServiceInstanceForBrowserContext \
+  BuildServiceInstanceForBrowserContext_ChromiumImpl
+#define GetForProfile GetForProfile_ChromiumImpl
+
+#include "src/chrome/browser/password_manager/android/login_db_deprecation_runner_factory.cc"
+
+#undef GetForProfile
+#undef BuildServiceInstanceForBrowserContext
+
+std::unique_ptr<KeyedService>
+LoginDbDeprecationRunnerFactory::BuildServiceInstanceForBrowserContext(
+    content::BrowserContext* context) const {
+#if BUILDFLAG(USE_LOGIN_DATABASE_AS_BACKEND)
```

Comment:
I'm confused about what is going on here. I assume eventually this flag will be removed? Why are we not doing the same thing as upstream here?

Reply:
Upstream uses OS based password manager now, we decided to keep using browser based PM. Once flag is removed, we intend to retain the code (move it to `brave-core` repo).

Reply:
why do we have this flag at all then if we always want the behavior to be nullptr return?

Reply:
Without this `#if` there is `code will never be executed` error for the upstream's code. I'll update the comment with more info.

Reply:
Added fixup with comment https://github.com/brave/brave-core/commit/1cf9b8382d2380082844d557082aef272cb92f0b

Reply:
I'm not following, if we never want to use `LoginDbDeprecationRunnerFactory` then it doesn't make any sense to have the buildflag here

Reply:
also if we never intend to use `LoginDbDeprecationRunner` we should just override the entire cc file with a no-op implementation

Reply:
Shouldn't we be overriding `use_login_database_as_backend` to true in config.js?

Reply:
As discussed in DM, this code was removed in favor of upstream's direct patch from `C134` https://github.com/brave/brave-core/commit/a6c34ebe3c107af648cdb15cc28236f1fd736681